### PR TITLE
Implement late-stage election status tracker.

### PR DESCRIPTION
**Pull request by YoEight**
Added: Inauguration status tracker

### CODE REVIEW

File: src/EventStore.Core.XUnit.Tests/Startup/StartupStatusTrackerTests.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+namespace EventStore.Core.XUnit.Tests.Startup {
+	public class StartupStatusTrackerTests {
+		private readonly FakeClock _clock = new();
+		private readonly StatusMetric _metric;
+		private readonly StartupStatusTracker _sut;
+
+		public StartupStatusTrackerTests() {
+			_metric = new StatusMetric(
+				new Meter($"Eventstore.Core.XUnit.Tests.{nameof(StartupStatusTrackerTests)}"),
```
Comment (timothycoleman) [inline src/EventStore.Core.XUnit.Tests/Startup/StartupStatusTrackerTests.cs:18]:
```suggestion
				new Meter($"{typeof(StartupStatusTrackerTests)}"),
```

i think i changed to doing it like this in my own tests somewhere along the way

File: src/EventStore.Core/Services/VNode/InaugurationManager.cs [hunk 2]
```diff
@@ -30,6 +30,7 @@ public class InaugurationManager :
 		private int _currentEpochNumber;
 		private long _replicationCheckpointTarget;
 		private long _indexCheckpointTarget;
+		private readonly IStartupStatusTracker _startupTracker;
```
Comment (timothycoleman) [inline src/EventStore.Core/Services/VNode/InaugurationManager.cs:33]: neater to move this up with the other readonly fields

File: src/EventStore.Core/Telemetry/StatusMetric.cs [hunk 3]
```diff
@@ -22,7 +22,7 @@ public class StatusMetric {
 			}
 		}
 
-		private IEnumerable<Measurement<long>> Observe() {
+		public IEnumerable<Measurement<long>> Observe() {
```
Comment (timothycoleman) [inline src/EventStore.Core/Telemetry/StatusMetric.cs:25]: can keep this private and use the TestMeterListener in the tests instead

File: src/EventStore.Core/Services/VNode/InaugurationManager.cs (hunk 2 above)
Comment (YoEight) [reply src/EventStore.Core/Services/VNode/InaugurationManager.cs]: Done

File: src/EventStore.Core.XUnit.Tests/Startup/StartupStatusTrackerTests.cs (hunk 1 above)
Comment (YoEight) [reply src/EventStore.Core.XUnit.Tests/Startup/StartupStatusTrackerTests.cs]: Done

**Review by hayley-jean — CHANGES_REQUESTED** [review]
Could you add a changelog message and description to the PR please?

**Comment by timothycoleman** [comment]
I've pushed some suggestions, if you're good with the changes we can squash them and i'll approve

**Review by timothycoleman — APPROVED** [review]
fab 👍

**Review by hayley-jean — APPROVED** [review]

**Pull request merged**